src/pydantic_avro/base.py

```diff
@@ -153,14 +169,26 @@ def get_fields(s: dict) -> List[dict]:
 
                 if key not in required:
                     if type(avro_type_dict["type"]) is list:
-                        avro_type_dict["type"].insert(0, "null")
+                        if "null" not in avro_type_dict["type"]:
+                            avro_type_dict["type"].insert(0, "null")
                     elif avro_type_dict.get("default") is None:
                         avro_type_dict["type"] = ["null", avro_type_dict["type"]]
-                        avro_type_dict["default"] = None
+                    avro_type_dict.setdefault("default", None)
+
+                # null must be the first element in the list
+                if type(avro_type_dict["type"]) is list and "null" in avro_type_dict["type"]:
+                    avro_type_dict["type"].remove("null")
+                    avro_type_dict["type"].insert(0, "null")
 
                 fields.append(avro_type_dict)
             return fields
 
         fields = get_fields(schema)
 
-        return {"type": "record", "namespace": namespace, "name": schema["title"], "fields": fields}
+        avro_schema_dict = {}
+        if cls.__doc__:
+            avro_schema_dict["doc"] = cleandoc(cls.__doc__)
```

Comment:
`if cls.__doc__` already acts as the necessary check for whether or not `schema['description']` both exists and wasn't inherited. 

So simply:
```suggestion
        if cls.__doc__:
            avro_schema_dict["doc"] = schema["description"]
```
Would work as desired~

Reply:
You're right, but if we use schema["description"] directly, it will get the docstring of the parent class if the child class's docstring isn't defined. 
[pydantic uses inspect.getdoc()](https://github.com/pydantic/pydantic/blob/v1.10.12/pydantic/schema.py#L578-L580).
So if we don't define a docstring, we'll get AvroBase's docstring.
That's why i used cls.\_\_doc\_\_ directly instead of using schema["description"]. 

pydantic V2 doesn't have this problem. It only gets the docstring of the child class.

Reply:
My point is,`cls.__doc__` will be `None` for both V1 and V2 if not directly defined. 
`__doc__` is not inherited in either case, and only the description is inherited in V1.
So if `__doc__` is defined, there's no reason to redo what pydantic does, and if `__doc__` isn't defined the line gets skipped in either case...